rust/src/bin/loadtest/main.rs

```diff
@@ -0,0 +1,88 @@
+mod flock;
+mod pets;
+mod stats;
+mod worker;
+
+use std::sync::Arc;
+
+use log::*;
+use scylla::Session;
+use structopt::StructOpt;
+
+use care_pet::{db, Result};
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "loadtest")]
+struct App {
+    // number of the startPets owners
+    #[structopt(long, default_value = "100")]
+    owners: usize,
+
+    // number of startPets to simulate
+    #[structopt(long, default_value = "100")]
+    pets: usize,
+
+    // number of sensors per pet
+    #[structopt(long, default_value = "4")]
+    sensors: usize,
+
+    // an interval between sensors measurements
+    #[structopt(long, default_value = "1s")]
+    interval: humantime::Duration,
+
+    // just write random data
+    #[structopt(long)]
+    writer: bool,
+
+    // number of parallel writers: Default: num of CPUs
+    #[structopt(long)]
+    workers: Option<usize>,
+
+    #[structopt(flatten)]
+    db_config: db::Config,
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    care_pet::log::init();
+
+    let app: App = App::from_args();
+    debug!("Configuration = {:?}", app);
+
+    info!("Welcome to the Pets simulator");
+    rlimit::utils::increase_nofile_limit(102400).unwrap();
```

Comment:
Panicking sounds a little harsh if you're not able to bump the file limit. Do you know why it needs to be bumped? Is it related to the number of connections? Or perhaps it just was there in the reference implementation in Go? Perhaps it's worth checking if the program also works without the bump, or just make it best-effort - try to increase the limit, but emit a warning and continue if it cannot be done.